### DISCUSSION TITLE
Blueprint Big Book of Data Analysis - 5  Advanced / Aspirational Workflow

### DISCUSSION THREAD

**Discussion by giscus** [context-dependent/bp-big-book#13] [General]
# Blueprint Big Book of Data Analysis - 5  Advanced / Aspirational Workflow

https://context-dependent.github.io/bp-big-book/wf-goal.html

**Comment by mehtavibha** [comment]
To make sure I'm understanding what we intend for folks to capture in the monitoring and decision logs, would an example of something that is captured in the monitoring log be how an issue of multiple/duplicate responses was handled? And for the decision log, an example could be not to go further with subgroup analysis after looking at it in the exploratory stage?

**Reply by context-dependent** [reply]
The purpose of the Monitoring Log is to document, in an obvious and durable place, things that happened over the course of data collection (e.g. your hypothetical of duplicate responses, some batch of survey links was messed up in some way, we needed to change the wording of a specific question, the approach to delivering or incentivizing surveys changed to address low response rates) 

The purpose of the Decision Log is similar, but its focus is on choices made during exploratory analysis and data cleaning (e.g. this critical demographic field has a lot of missing values, so we imputed replacements for those missing values in way X; after discussions with person Y, we determined that value Z in field M indicates that a participant didn't end up attending any classes, so we exclude them from the analysis).

It is important for the integrity and efficiency of the analysis that relevant events and course corrections identified in the monitoring process are recorded. The same is true of the questions, answers, and decisions that end up reflected in the final shape of the research dataset. So long as their contents are obviously located and accessibly structured I am personally agnostic wrt the form they take. 

I encourage you to check out the Issues section of the fsc-benchmarks repo Alex has invited you to for an example of the 'Decision Log' concept in action.

**Reply by context-dependent** [reply]
Also, had this [drafty workflow pilots section](https://context-dependent.github.io/bp-big-book/wf-pilots.html) written up but not indexed until just now. 

Hoping to use that to explain and advertise promising workflow practices happening around Blueprint.

**Comment by mehtavibha** [comment]
If the data is ready for analysis at the exploratory stage, what is the "iteratively build data pipeline" step under AP2?

**Reply by context-dependent** [reply]
We might differ on our definitions of exploratory analysis. My concept of of the term aligns with Andrew Heiss': 

> Exploratory Data Analysis (EDA) is a data-driven approach to analysis which aims to expose the properties and structure of a dataset, and from here suggest directions for analytic inquiry. In an EDA, relationships are quickly inferred, anomalies labelled, assumptions tested and new hypotheses and ideas are formulated. EDA relies heavily on visual approaches to analysis – it is common to generate many dozens of (often throwaway) data graphics. (from his course [Computational Methods for Social Science](https://www.roger-beecham.com/comp-sds/class/04-class/)

The data pipeline is the reproducible process the data analyst builds to transform the raw input data into the final research dataset. EDA is the process by which the data analyst learns what that the pipeline can and should do to make that happen.

**Comment by mehtavibha** [comment]
What exactly happens at the exploratory stage of analysis?

**Reply by context-dependent** [reply]
see my reply above.

**Comment by mehtavibha** [comment]
I know we don't want to provide templates but I wonder if we can consider providing examples of the different artefacts we recommend, just so people have a sense of what should ideally go into them.

**Reply by context-dependent** [reply]
Definitely good, though we don't really have those examples to provide (at least of `APX`s). For that reason, I'd suggest that we focus on rolling out the git / github workflow and the repo issues decision log first, along with our general intention to increase 'planfulness' and improve documentation in the analysis process.

**Comment by mehtavibha** [comment]
Another hook for tools, rules and standards is survey design. For e.g., using validated scales, a standard set of questions to capture socioeconomic characteristics, or employment-related indicators.

**Reply by context-dependent** [reply]
Absolutely. I'd put 'How do we work with validated scales?' on the agenda for our Stats Summit.
